src/usr/local/buildpack/utils/folders.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/bash
+
+# Constants
+ROOT_DIR=/opt/buildpack
+ROOT_UMASK=750
+USER_UMASK=770
+
+# Will set up the general folder structure for the whole buildpack installation
+# This will take $TEST_ROOT into account, so can also be used during testing
+function setup_directories () {
+  local install_dir
+  install_dir=$(get_install_dir)
+
+  mkdir -p "${install_dir}"
+  # contains the installed tools
+  mkdir -m 770 "$(get_tools_path)"
+  # contains env for the installed tools
+  mkdir -m 770 "$(get_env_path)"
+  # contains the latest version of the tools
+  mkdir -m 770 "$(get_version_path)"
+  # contains the wrapper and symlinks for the tools
+  mkdir -m 770 "$(get_bin_path)"
+}
+
+# Creates the given folder path with root and user umask depending on the caller
+# Will also create intermediate folders with correct umask
+function create_folder () {
+  local folder=${1}
+  check folder
+
+  local parent
+  parent=$(dirname "${folder}")
+
+  if [ -d "${folder}" ]; then
+    return
+  fi
+
+  if [ ! -d "${parent}" ]; then
+    create_folder "$parent"
+  fi
+
+  local umask=$USER_UMASK
+  if [ "$(is_root)" -eq 0 ]; then
```

Comment:
`is_root`? Do we need this function?

Reply:
It is needed to make testing possible. 
I also changed the `utils.sh` but I am struggling to make commits that will not brake anything.
So locally it is already changed and it makes sense as soon as I push more code :)

Reply:
https://github.com/containerbase/buildpack/pull/269/files#diff-a968fbd6e0e6b76f847072007c8b5d39bfcbd8f06e8d8663007b394bf131a3ecR45-R48

Here you see it is needed to test the files with "root" id and without